jupyter-extension/jupyterlab_pachyderm/handlers.py

```diff
@@ -364,20 +387,52 @@ async def get(self):
 
 
 class AuthLoginHandler(BaseHandler):
-    async def get_token(self, oidc_state: str):
-        token = self.client.auth.authenticate(oidc_state=oidc_state).pach_token
-        self.settings["pachyderm_client"].auth_token = token
-        self.settings["pfs_contents_manager"] = PFSManager(client=self.client)
-        self.settings["datum_contents_manager"] = DatumManager(client=self.client)
-        self.settings["pachyderm_pps_client"] = PPSClient(client=self.client)
 
     @tornado.web.authenticated
     async def put(self):
         try:
+            # Note: The auth workflow is for the backend to initiate the process by
+            # calling auth.get_oidc_login() which returns a url for the user to login
+            # with, and a state token that the backend can use to complete the workflow.
+            # Therefore, we send the url to the frontend for the user to login with and
+            # wait until the server has created a new session token, which we then retrieve.
             oidc_response = self.client.auth.get_oidc_login()
-            asyncio.create_task(self.get_token(oidc_response.state))
             response = oidc_response.to_json()
-            self.finish(response)
+            await self.finish(response)
+
+            # Unfortunately, there doesn't seem to be a way for us to use the synchronous
```

Comment:
do you think it is feasible to extend the sdk to implement an async `auth.authenticate`?

Reply:
yes potentially. I've spent some time trying to think how best to do that, and I'm not sure what the best method would be for maintainability. I think we might want to wait on the implementation of this until we find a few more required asynchronous methods